xwiki-pro-macros-api/src/main/java/com/xwiki/macros/tab/internal/TabMacro.java

```diff
@@ -0,0 +1,164 @@
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.rendering.block.GroupBlock;
+import org.xwiki.rendering.block.MetaDataBlock;
+import org.xwiki.rendering.macro.MacroContentParser;
+import org.xwiki.rendering.macro.MacroExecutionException;
+import org.xwiki.rendering.macro.descriptor.DefaultContentDescriptor;
+import org.xwiki.rendering.parser.Parser;
+import org.xwiki.rendering.syntax.Syntax;
+import org.xwiki.rendering.syntax.SyntaxType;
+import org.xwiki.rendering.transformation.MacroTransformationContext;
+import org.xwiki.skinx.SkinExtension;
+import org.xwiki.text.StringUtils;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xwiki.macros.AbstractProMacro;
+import com.xwiki.macros.tab.macro.TabMacroParameters;
+import com.xwiki.macros.tab.macro.TransitionEffect;
+
+/**
+ * Tab macro: a tab element for the tab-group macro.
+ *
+ * @version $Id: $
+ * @since 1.24.0
+ */
+@Component
+@Named("tab")
+@Singleton
+public class TabMacro extends AbstractProMacro<TabMacroParameters>
+{
+    private static final String NAME = "Tab";
+
+    private static final String DESCRIPTION =
+        "Tab element for the tab group macro. This is expected to be put as the tab-group macro content.";
+
+    private static final String CONTENT_DESCRIPTION = "The content to be displayed in the tab.";
+
+    private static final String BLOCK_PARAM_CLASS = "class";
+
+    @Inject
+    protected MacroContentParser contentParser;
+
+    @Inject
+    private Provider<XWikiContext> xwikiContextProvider;
+
+    @Inject
+    private Logger logger;
+
+    @Inject
+    @Named("context")
+    private Provider<ComponentManager> componentManagerProvider;
+
+    @Inject
+    @Named("ssrx")
+    private SkinExtension ssrx;
+
+    /**
+     * Create and initialize the descriptor of the macro.
+     */
+    public TabMacro()
+    {
+        super(NAME, DESCRIPTION,
+            new DefaultContentDescriptor(CONTENT_DESCRIPTION, true, Block.LIST_BLOCK_TYPE),
+            TabMacroParameters.class);
+    }
+
+    @Override
+    public boolean supportsInlineMode()
+    {
+        return false;
+    }
+
+    @Override
+    protected List<Block> internalExecute(TabMacroParameters parameters, String content,
+        MacroTransformationContext context) throws MacroExecutionException
+    {
+        // Don't show tab in edit mode.
+        Syntax syntax = context.getTransformationContext().getTargetSyntax();
+        SyntaxType targetSyntaxType = syntax == null ? null : syntax.getType();
+        if (SyntaxType.ANNOTATED_HTML.equals(targetSyntaxType) || SyntaxType.ANNOTATED_XHTML.equals(targetSyntaxType)) {
+            // Handle edit mode
+            ssrx.use("css/tabmacro.css");
+            List<Block> tabLabelBlock;
+            try {
+                Parser parser = componentManagerProvider.get().getInstance(Parser.class, "xwiki/2.1");
```

Comment:
well, I just use a parser here because we can't just create a word block with the text as we need an other block type for special char. So in this case should we use the `plain/1.0` parser ?

Reply:
@michitux, what do you think ?

Reply:
Yes, otherwise, text like `**label**` will be rendered as `<strong>label</strong>`.